### PR TITLE
Increments wal corruption metric on error during checkpointing

### PR DESCRIPTION
**Pull request by Harkishen-Singh**
Signed-off-by: Harkishen-Singh <harkishensingh@hotmail.com>

Fixes: #7443

### CODE REVIEW

**Comment by roidelapluie** [comment]
should we have a labe on those metrics? We have WAL, MMAP-head, checkpoints..

**Comment by brian-brazil** [comment]
That's probably over-complicating things. A WAL corruption is a WAL corruption.

File: tsdb/head.go [hunk 1]
```diff
@@ -837,6 +837,7 @@ func (h *Head) Truncate(mint int64) (err error) {
 	h.metrics.checkpointCreationTotal.Inc()
 	if _, err = wal.Checkpoint(h.wal, first, last, keep, mint); err != nil {
 		h.metrics.checkpointCreationFail.Inc()
+		h.metrics.walCorruptionsTotal.Inc()
```
Comment (codesome) [inline tsdb/head.go:840]: Checkpointing could fail for various reasons, we should increment only when it's a WAL corruption. There is `wal.CorruptionErr` which you could check for (don't forget `errors.Cause`).

**Comment by Harkishen-Singh** [comment]
@codesome done! I hope that's what we want.

**Pull request merged**